google-cloud-bigquery/lib/google/cloud/bigquery/service.rb

```diff
@@ -285,6 +285,56 @@ def delete_model dataset_id, model_id
           execute { service.delete_model @project, dataset_id, model_id }
         end
 
+        ##
+        # Creates a new routine in the dataset.
+        def insert_routine dataset_id, new_routine_gapi
+          execute { service.insert_routine @project, dataset_id, new_routine_gapi }
+        end
+
+        ##
+        # Lists all routines in the specified dataset.
+        # Requires the READER dataset role.
+        # Unless readMask is set in the request, only the following fields are populated:
+        #   etag, projectId, datasetId, routineId, routineType, creationTime, lastModifiedTime, and language.
+        def list_routines dataset_id, options = {}
```

Comment:
Perhaps `options` should be a kwargs, for cleaner Ruby 3 behavior?

```ruby
def list_routines dataset_id, **options
```

Reply:
Usages of `options = {}` in this gem predate Ruby 2.0+ keyword arguments, so I'll update to those where appropriate and otherwise to `**kwargs`.

Reply:
Fixed everywhere in the gem.

Reply:
Oh. I was thinking just for the new code. The various `from_options` calls all suggest they're creating things "from" an options hash, and anyway they're private so I thought they weren't that important. However, if you do want to do them: (1) there's another example in `QuerySet::Updater.from_options`, and (2) you should update the call sites as well because now options hashes have to be double-splatted when passed in.

Reply:
Agreed. After seeing how this turned out (the `options` hashes are sometimes way too big to inline as keyword args), I restored `CopyJob.from_options` and `ExtractJob.from_options` from `**kwargs` to `options` hash.